test/extended/images/mongodb_ephemeral.go

```diff
@@ -24,8 +24,13 @@ var _ = g.Describe("[images][mongodb] openshift mongodb image", func() {
 			g.By("creating a new app")
 			o.Expect(oc.Run("new-app").Args("-f", templatePath).Execute()).Should(o.Succeed())
 
-			g.By("expecting the mongodb service get endpoints")
-			o.Expect(oc.KubeFramework().WaitForAnEndpoint("mongodb")).Should(o.Succeed())
+			g.By("waiting for the deployment to complete")
+			exutil.WaitForADeployment(
+				oc.KubeREST().ReplicationControllers(oc.Namespace()),
```

Comment:
Not sure if it is worth with but... maybe you could call `oc.Namespace()` only once within a test?
I believe that function call will translate to `oc project` or something like that?

Right now there are two calls to `oc.Namespace()`.